src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/XamlLanguageServerProtocol.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.Xaml;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Roslyn.Utilities;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.VisualStudio.LanguageServices.Xaml.LanguageServer
+{
+    /// <summary>
+    /// Implements the Language Server Protocol for XAML
+    /// </summary>
+    [Shared]
+    [Export(typeof(XamlLanguageServerProtocol))]
+    internal sealed class XamlLanguageServerProtocol : ILanguageServerProtocol
```

Comment:
Could we potentially just make an abstract class for ILanguageServerProtocol instead of an interface?  Then we could share the map creation and ExecuteRequestAsync code

If while you're doing it you could change the name to something like AbstractRequestHandlerProvider that would be great as well (the old file went through a couple refactorings and never got appropriately renamed)

Reply:
Good idea, thanks.

Reply:
:+1:

This would help us out too (which is presumably why David mentioned it :))

Reply:
And done too. :-)